permission/core/cache.go

```diff
@@ -579,16 +600,11 @@ func CheckIfAdminAccount(acctId common.Address) bool {
 }
 
 // validates if the account can transact from the current node
-func ValidateNodeForTxn(hexnodeId string, from common.Address) bool {
-	if !PermissionsEnabled() || hexnodeId == "" {
+func ValidateNodeForTxn(nodeId enode.ID, from common.Address) bool {
+	if !PermissionsEnabled() || nodeId == emptyNodeId {
```

Comment:
so back with what I don't like, maybe an `IsEmpty()` function might work or a `len(nodeId) == 0`?

Reply:
From my point of view function is too complex for this action and I didn't want to change enode logic for this case. 

`len(nodeId) == 0` will not work for this case because enode.ID is fixed size bytes array and `len(enode.ID{})` is 32 bytes

My solution is to compare node id with empty enode.ID value . Please check this fix: https://github.com/ConsenSys/quorum/pull/1578/commits/39fb51961fa51af64d0768e0b05c681a1a3b6d42#diff-a39234da6bbd13a5c02519dd2dff8aac2ea8a8d948ac58ed60c4ce5bd84f34b5R603